web/html/src/manager/admin/setup/products/products.js

```diff
@@ -118,6 +119,11 @@ class ProductsPageWrapper extends React.Component {
           scheduleResyncItems: [],
           scheduledItems: [],
         });
+        if (currentObject.state.noToolsChannelSubscription && currentObject.state.issMaster) {
```

Comment:
It is better to set the state everything at once. I'd just move these lines right before `setState` few lines above and preparing a variable to populate with messages. Something like the following:
```js
let resultMessages = [];
      if (currentObject.state.noToolsChannelSubscription && currentObject.state.issMaster) {
        resultMessages = MessagesUtils.warning(t("No SUSE Manager Server Subscription availabel. Products requiring Client Tools Channel will not be shown."))
      }
currentObject.setState({
          serverData: data[_DATA_ROOT_ID],
          errors: resultMessages,
          loading: false,
          selectedItems: [],
          scheduleResyncItems: [],
          scheduledItems: [],
        });
```

Reply:
done